src/goto-instrument/contracts/cfg_info.h

```diff
@@ -167,13 +167,27 @@ class loop_cfg_infot : public cfg_infot
       return true;
   }
 
+  /// Recursively check whether `expr` contains a local symbol.
+  bool contain_locals(const exprt &expr)
+  {
+    if(expr.id() == ID_symbol)
+      return is_local(to_symbol_expr(expr).get_identifier());
+
+    for(const auto &sub_e : expr.operands())
+    {
+      if(contain_locals(sub_e))
+        return true;
+    }
+
+    return false;
+  }
```

Comment:
May I suggest that you use `find_symbol_identifiers` rather than rolling your own search here?